### DISCUSSION TITLE
is the api able to stream  audio?

### DISCUSSION THREAD

**Discussion by Tsamsiyu1** [advplyr/audiobookshelf#3067] [Q&A] [answered]
I read myself into the API and tryed some stuff. it worked well. i got back all the json data i wanted. but then i tryed to play an item. and i get back a json data with all the data of the item and the device i am playing from but i dont get an audio stream. Am I just stupid or what am i doing wrong?

this is my code:
`import requests
from pydub import AudioSegment
from pydub.playback import play
from io import BytesIO

def main():
    # Konfiguration
    api_token = "my token"
    server = "my server address"
    endpoint = "/api/items/1f97422c-3ed8-4237-bdc5-52463f8a5c86/play"
    
    # URL-Zusammensetzung
    url = f"{server}{endpoint}"
    
    # Header mit dem API-Token
    headers = {
        "Authorization": f"Bearer {api_token}",
        "Content-Type": "application/json"
    }
    
    data = {
        "deviceInfo": {"clientVersion": "0.0.1"},
        "supportedMimeTypes": ["audio/mpeg"]  # MP3 auswählen
    }

    try:
        response = requests.post(url, headers=headers, json=data, stream=True)

        if response.status_code == 200:
            content_type = response.headers.get('Content-Type')
            print(f"Content-Type: {content_type}")

            if content_type and "audio" in content_type:
                print("Audio-Stream erfolgreich abgerufen")
                audio_data = BytesIO(response.content)
                audio = AudioSegment.from_file(audio_data, format="mp3")
                play(audio)
            else:
                print("no audio file:", response.text)
        else:
            print("error on the stream:", response.status_code, response.text)
    except requests.ConnectionError as e:
        print(f"error when trying to connect to {url}: {e}")

if __name__ == "__main__":
    main()`

would be nice if anybody has some awnsers for me.

**Comment by advplyr** [comment, marked as answer]
Check out https://api.audiobookshelf.org/#play-a-library-item-or-podcast-episode

That endpoint returns an object that includes an array of `audioTracks` that include a `contentUrl`

**Reply by Tsamsiyu1** [reply]
thanks this means i was just dump.
